processing/src/main/java/io/druid/query/groupby/epinephelinae/Grouper.java

```diff
@@ -231,6 +235,18 @@ public String toString()
     KeyComparator bufferComparator();
 
     /**
+     * When pushing down limits, it may also be necessary to compare aggregated values along with the key
+     * using the bufferComparator.
+     *
+     * @param aggregatorFactories Array of aggregators from a GroupByQuery
+     * @param aggregatorOffsets Offsets for each aggregator in aggregatorFactories pointing to their location
+     *                          within the grouping key + aggs buffer.
+     *
+     * @return comparator for keys + aggs
+     */
+    KeyComparator bufferComparatorWithAggregators(AggregatorFactory[] aggregatorFactories, int[] aggregatorOffsets);
```

Comment:
I renamed this to `BufferComparator`.

I think making a `BufferComparatorFactory` would be problematic for the `RowBasedKeySerde`, the comparisons on dimensions rely on the dictionary built by the KeySerde, there would be a lot of redundant information between the BufferComparatorFactory and the KeySerde (e.g., the serdeHelpers, timestamp handling information, dictionaries). 

I suppose I view KeySerde as a sort of BufferComparatorFactory, and I feel like the additional withAggregators comparator method isn't inappropriate. 

It's generating a comparator that deals with serialized key portions, interleaved with some extra aggregator comparisons, so the KeySerde is still dealing with its core function of handling operations related to serialized keys.